mediacapture-streams/idlharness.https.window.js

```diff
@@ -5,50 +5,44 @@
 
 // https://w3c.github.io/mediacapture-main/
 
-
-promise_test(async () => {
-  const srcs = ['mediacapture-streams','dom','html'];
-  const [idl, dom, html] = await Promise.all(
-    srcs.map(i => fetch(`/interfaces/${i}.idl`).then(r => r.text())));
-
-  const idl_array = new IdlArray();
-  idl_array.add_idls(idl);
-  idl_array.add_dependency_idls(html);
-  idl_array.add_dependency_idls(dom);
-
-  const devices = [];
-  // Errors will be surfaced in idlharness.js's test_object below.
-  try {
-    const list = await navigator.mediaDevices.enumerateDevices();
-    for (const item of list) {
-      switch (item.kind) {
-      case 'audioinput':
-      case 'videoinput':
-      case 'audiooutput':
-        self[item.kind] = item;
-        devices.push(item.kind);
-      default:
-        assert_unreached(
-          'media.kind should be one of "audioinput", "videoinput", or "audiooutput".');
+idl_test(
+  ['mediacapture-streams'],
+  ['dom', 'html'],
+  async idl_array => {
+    const inputDevices = [];
```

Comment:
There's some refactoring here to distinguish `InputDeviceInfo` and `MediaDeviceInfo` which has led to new subtests.

Reply:
Ah, I hadn't looked at the history of how it came to be this way.